File: java8/src/java8/Main04.java
package java8;

import java.util.Arrays;
import java.util.List;

/** Example of lambdas with method handles. */
public class Main04 {

  public static void main(String[] args) {
    List<String> list = Arrays.asList("1", "2", "3", "44");

    list.forEach(System.out::println);

    Logic l = new Logic();
    list.forEach(l::println);
  }

  public static class Logic {
    public void println(String s) {
      System.out.println("string: " + s);
    }
  }
}

File: java8-scala/src/java8/Main04.scala
package java8

/** An example of lambdas and method handles. */
object Main04 {
  def main(args: Array[String]) {
    val list = Seq("1", "2", "3", "44")
    
    list.foreach(println)

    val l = new Logic()
    list.foreach(l.println)
  }
  
  class Logic {
    def println(s: String) = System.out.println("string: " + s)
  }
}